src/service/resources-impl.js

```diff
@@ -1008,7 +1008,7 @@ export class Resources {
       // Note that when the document is not visible, neither are any of its
       // elements to reduce CPU cycles.
       const shouldBeInViewport = (this.visible_ && r.isDisplayed() &&
-          r.overlaps(visibleRect));
+          (r.isFixed() || r.overlaps(visibleRect)));
```

Comment:
I'd prefer that too. The problem with this is, essentially, the `layoutRect` is implemented completely incorrectly for fixed elements. And comparing overlap here with `visibleRect` is also incorrect. This functionality has to be refactored in each separate layer, which is a lot bigger fix (#3434).